Recipes/definitions.recipe.json

```diff
@@ -2087,6 +2227,141 @@
+                  "DefaultValue": "three"
+                },
+                "ContentIndexSettings": {}
+              }
+            }
+          ]
+        },
+        {
+          "Name": "ContentFeed",
+          "Settings": {},
+          "ContentPartFieldDefinitionRecords": [
+            {
+              "FieldName": "NumericField",
+              "Name": "PageSize",
+              "Settings": {
+                "ContentPartFieldSettings": {
+                  "DisplayName": "Number of Items",
+                  "Position": "3"
+                },
+                "NumericFieldSettings": {
+                  "Hint": "Number of items to display (per page, if paging is enabled)",
+                  "Minimum": 0.0
+                },
+                "ContentIndexSettings": {}
+              }
+            },
+            {
+              "FieldName": "BooleanField",
+              "Name": "Paging",
+              "Settings": {
+                "ContentPartFieldSettings": {
+                  "DisplayName": "Paging",
+                  "Position": "4"
+                }
+              }
+            },
+            {
+              "FieldName": "BooleanField",
+              "Name": "HideWidgetHeader",
+              "Settings": {
+                "ContentPartFieldSettings": {
+                  "DisplayName": "Hide Widget Header",
+                  "Position": "0"
+                },
+                "BooleanFieldSettings": {
+                  "Hint": "By default, this widget displays the title and links in a header area. Tick this box to hide the header."
+                },
+                "ContentIndexSettings": {}
+              }
+            },
+            {
+              "FieldName": "QueryField",
+              "Name": "Items",
+              "Settings": {
+                "ContentPartFieldSettings": {
+                  "DisplayName": "Items",
+                  "Position": "1"
+                }
+              }
+            },
+            {
+              "FieldName": "TextField",
+              "Name": "Columns",
+              "Settings": {
+                "ContentPartFieldSettings": {
+                  "DisplayName": "Columns",
+                  "Editor": "PredefinedList",
+                  "Position": "2"
+                },
+                "TextFieldSettings": {
+                  "Hint": "Specify how many columns to display the items in on desktop devices"
+                },
+                "TextFieldPredefinedListEditorSettings": {
+                  "Options": [
+                    {
+                      "name": "Four",
+                      "value": "four"
+                    },
+                    {
+                      "name": "Three",
+                      "value": "three"
+                    },
+                    {
+                      "name": "Two",
+                      "value": "two"
+                    },
+                    {
+                      "name": "One",

```

Comment:
I've been wondering about the "Columns" device we've introduced on Sumo - should it perhaps be a Part? Like "Columnable", which adds column related fields like this one (and possibly others in future, like if we do gutter settings, or a separate field for number of columns on mobile, or anything like that)

In terms of default output, we could then output the value in two ways on elements:

`{block}--{columnValue}-columns columns--{columnValue}`

So in practice, the output might look like:
`<div class="tiles tiles--three-columns columns--three">`

This would enable us to set up some sensible defaults for columns as a set of columns helper classes in CSS, but still allow the flexibility for someone to say "Actually I want tiles specifically to behave slightly different at this breakpoint" (which did actually happen on Sumo, we decided to force 2 columns on mobile)

The two places I see this being used at present are Tiles and on Content Feed

Reply:
Great suggestion.

Reply:
Need to apply this to the themes, however I assume this is what you had in mind - https://github.com/EtchUK/Etch.OrchardCore.Widgets/pull/49/commits/1cb15e41c6bda4d2f17e963219bbf22b6258a0e1

Reply:
All done.